### PR TITLE
logger: improve function create_prefix

### PR DESCRIPTION
**Pull request by xiekeyang**
This improves a little creating prefix. Generate prefix items using `if in let` expression, and join all ones by separator `:`, as to make easy to add or delete item from/to prefix.

Signed-off-by: xiekeyang <keyang.xie@gmail.com>

## Reason for This PR

This improves a little on creating prefix in logger.

## Description of Changes

1. To use `if in let` expression as to simplify the condition block.

2. To organize ones to vector and join them by separator `:`, as to make easy to add or delete item from/to prefix.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided.
- [x] The description of changes is clear and encompassing.
- [x] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [x] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [x] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [x] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.
- [x] Either no new `unsafe` code has been added, or, the newly added `unsafe`
      code is unavoidable and properly documented.

### CODE REVIEW

**Comment by dianpopa** [comment]
Hi  @xiekeyang ,

Thanks for the PR. Seems that the log-related integration tests are failing the newly added prefix-creating logic.
You can debug the failures by doing the below (only after [you have setup docker](https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md#appendix-b-setting-up-docker)):
```
cd tools
./devtool shell -p
cd tests
pytest integration_tests/functional/test_logging.py
```

**Comment by dianpopa** [comment]
> Hi @xiekeyang ,
> 
> Thanks for the PR. Seems that the log-related integration tests are failing the newly added prefix-creating logic.
> You can debug the failures by doing the below (only after [you have setup docker](https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md#appendix-b-setting-up-docker)):
> 
> ```
> cd tools
> ./devtool shell -p
> cd tests
> pytest integration_tests/functional/test_logging.py
> ```

If you need help, you can ping me on slack.

**Comment by xiekeyang** [comment]
> Hi @xiekeyang ,
> 
> Thanks for the PR. Seems that the log-related integration tests are failing the newly added prefix-creating logic.
> You can debug the failures by doing the below (only after [you have setup docker](https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md#appendix-b-setting-up-docker)):
> 
> ```
> cd tools
> ./devtool shell -p
> cd tests
> pytest integration_tests/functional/test_logging.py
> ```

@dianpopa Thanks so much! I noticed the failed cases, and would fix them ASAP.

**Comment by xiekeyang** [comment]
@dianpopa ,  all checks have passed, PTAL, thanks a lot!

File: src/logger/src/lib.rs [hunk 1]
```diff
@@ -390,33 +390,36 @@ impl Logger {
...
             }
-        }
+        } else {
+            "".to_string()
+        };
 
-        res.push_str("]");
-        res
+        let mut prefix: Vec<String> = vec![ins_id, level, pth, line];
+        prefix.retain(|i| !i.is_empty());
```
Comment (dianpopa) [inline src/logger/src/lib.rs]: :+1:

**Pull request merged**